app/src/main/java/it/ministerodellasalute/immuni/logic/exposure/ExposureAnalyticsManager.kt

```diff
@@ -0,0 +1,325 @@
+ */
+
+package it.ministerodellasalute.immuni.logic.exposure
+
+import android.util.Base64
+import androidx.annotation.VisibleForTesting
+import it.ministerodellasalute.immuni.extensions.attestation.AttestationClient
+import it.ministerodellasalute.immuni.extensions.nearby.ExposureNotificationManager
+import it.ministerodellasalute.immuni.extensions.notifications.PushNotificationManager
+import it.ministerodellasalute.immuni.extensions.utils.*
+import it.ministerodellasalute.immuni.logic.exposure.models.ExposureAnalyticsOperationalInfo
+import it.ministerodellasalute.immuni.logic.exposure.models.ExposureSummary
+import it.ministerodellasalute.immuni.logic.exposure.repositories.ExposureAnalyticsNetworkRepository
+import it.ministerodellasalute.immuni.logic.exposure.repositories.ExposureAnalyticsStoreRepository
+import it.ministerodellasalute.immuni.logic.exposure.repositories.ExposureReportingRepository
+import it.ministerodellasalute.immuni.logic.settings.ConfigurationSettingsManager
+import it.ministerodellasalute.immuni.logic.settings.models.ConfigurationSettings
+import it.ministerodellasalute.immuni.logic.user.UserManager
+import it.ministerodellasalute.immuni.logic.user.models.Province
+import java.security.MessageDigest
+import java.security.SecureRandom
+import java.util.*
+import kotlinx.coroutines.delay
+
+class ExposureAnalyticsManager(
+    private val networkRepository: ExposureAnalyticsNetworkRepository,
+    private val exposureReportingRepository: ExposureReportingRepository,
+    private val attestationClient: AttestationClient,
+    private val baseOperationalInfoFactory: () -> BaseOperationalInfo,
+    private val schedulerFactory: () -> Scheduler,
+    private val randomSaltFactory: () -> String = { randomSalt() },
+    private val base64Encoder: (ByteArray) -> String = { Base64.encodeToString(it, Base64.NO_WRAP) }
+) {
+    constructor(
+        storeRepository: ExposureAnalyticsStoreRepository,
+        settingsManager: ConfigurationSettingsManager,
+        networkRepository: ExposureAnalyticsNetworkRepository,
+        exposureReportingRepository: ExposureReportingRepository,
+        attestationClient: AttestationClient,
+        baseOperationalInfoFactory: () -> BaseOperationalInfo,
+        schedulerFactory: () -> Scheduler = {
+            Scheduler(
+                storeRepository = storeRepository,
+                settings = settingsManager.settings.value
+            )
+        }
+    ) : this(
+        networkRepository = networkRepository,
+        exposureReportingRepository = exposureReportingRepository,
+        attestationClient = attestationClient,
+        baseOperationalInfoFactory = baseOperationalInfoFactory,
+        schedulerFactory = schedulerFactory
+    )
+
+    companion object {
+        private fun randomSalt(): String {
+            val salt = ByteArray(16)
+            SecureRandom().nextBytes(salt)
+            return Base64.encodeToString(salt, Base64.NO_WRAP)
```

Comment:
The fact is that salt should be a string and the fact that we're using a Base64 encoding is an implementation detail

Reply:
Okay, that makes sense, but could we remove the field encoder then and use the same static function?
(So that there is no case of two different base64 encoders)

Reply:
added a private method `encodeToBase64String` that is used both by the field and the randomSalt